### PR TITLE
Notifications redux

### PR DESCRIPTION
**Pull request by andrewxhill**

### CODE REVIEW

File: App/Models/TextileTypes.ts [hunk 1]
```diff
@@ -22,6 +22,13 @@ export interface Devices {
   items: Device[]
 }
 
+export interface Notification {
```
Comment (asutula) [inline App/Models/TextileTypes.ts]: I originally made the `TextileTypes` file to keep track of the model types that are used with `textile-go`... Let's make a point of cleaning that up, makes it easier to deal with the framework update process.

File: App/Redux/RootReducer.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ export default combineReducers({
   contacts: contactsReducer,
   devices: devicesReducer,
   preferences: prefrencesReducer,
+  notifications: notificationsReducer,
```
Comment (asutula) [inline App/Redux/RootReducer.ts]: Cool, think you need to add the 'notifications' key to the persistence whitelist as well.

File: App/Models/TextileTypes.ts (hunk 1 above)
Comment (andrewxhill) [reply App/Models/TextileTypes.ts]: cool. pretty sure this guy will live here regardless, it's the UI-based reshaping of notifications. making it so i can easily style the feed list item by type of notification, stuff that go isn't going to care about

File: App/Redux/RootReducer.ts (hunk 2 above)
Comment (andrewxhill) [reply App/Redux/RootReducer.ts]: done

File: App/Models/TextileTypes.ts [hunk 3]
```diff
@@ -22,6 +22,13 @@ export interface Devices {
...
+  category: 'node' | 'devices' | 'threads' | 'content'
+  type: string
+  date: number
+  payload?: any
+}
```
Comment (asutula) [inline App/Models/TextileTypes.ts]: Seems like we'd want to track a 'read' or 'unread' status.

File: App/Redux/NotificationsRedux.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { createAction, ActionType, getType } from 'typesafe-actions'
+import * as TextileTypes from '../Models/TextileTypes'
+
+const actions = {
+  newNotification: createAction('NEW_NOTIFICATION_REQUEST', resolve => {
```
Comment (asutula) [inline App/Redux/NotificationsRedux.ts]: Good start I think. I could imagine some other actions we'd want:

1. Acknowledge notification - Fired it the user taps or opens a notification, toggles the status to 'read'
2. Acknowledge all notifications - Fired in a situation like, user opens the notifications screen and is there for at least 5 seconds, and then navigates away from the notifications screen. Marks all unread notifications as read.

**Pull request merged**